tests/lib/mock_backend.py

```diff
@@ -7,47 +7,72 @@
 from .lazy_orm import LazyBase, LazyColumn
 
 
-class MockBackend(BaseBackend):
-    backend_id = "mock"
-    query_engine_class = MssqlQueryEngine
-    patient_join_column = "PatientId"
-
-    patients = MappedTable(
-        source="patients",
-        columns=dict(
-            height=Column("float", source="Height"),
-            date_of_birth=Column("date", source="DateOfBirth"),
-        ),
-    )
-    practice_registrations = MappedTable(
-        source="practice_registrations",
-        columns=dict(
-            stp=Column("varchar", source="StpId"),
-            date_start=Column("date", source="StartDate"),
-            date_end=Column("date", source="EndDate"),
-        ),
-    )
-    clinical_events = MappedTable(
-        source="events",
-        columns=dict(
-            code=Column("varchar", source="EventCode"),
-            system=Column("varchar", source="System"),
-            date=Column("date", source="Date"),
-            result=Column("float", source="ResultValue"),
-        ),
-    )
-
-    positive_tests = QueryTable(
-        columns=dict(
-            patient_id=Column("integer"),
-            result=Column("boolean"),
-            test_date=Column("date"),
-        ),
-        query="""
-            SELECT PatientID as patient_id, PositiveResult as result, TestDate as test_date FROM all_tests
-        """,
-    )
-
+def backend_factory(query_engine_class):
+    """
+    Return a instance of MockBackend associated with the supplied Query Engine
+    """
+    # We have to give this variable a new name because Python's scoping rules won't let
+    # you write:
+    #
+    #    def test():
+    #       foo = "bar"
+    #       class Test:
+    #           foo = foo
+    #
+    # Although this works outside of a function for reasons I don't fully understand.
+    query_engine_class_ = query_engine_class
```

Comment:
Could you give the function argument the alternative name instead of aliasing via a local variable?

Reply:
I did do that originally, but I got irrationally annoyed by the fact that even positional argument names are arguably part of the public interface to a function (you can always call them as keywords if you want to) and now I'm forced to give my function a weirdly named argument just because of some Python scoping silliness. The grown up thing to do would be to acknowledge that this doesn't matter and call it `query_engine_cls` or something like that but I wasn't feeling very grown up at the time.

Reply:
As you wish. Code needs a _bit_ of character.

Reply:
I'll change it; I've calmed down a bit now.

Reply:
I still don't understand why this works:
```console
python3.9 -c '

foo = "bar"
class Test:
  foo = foo
print(Test.foo)
'
```

And this doesn't:
```console
python3.9 -c '

def test():
  foo = "bar"
  class Test:
    foo = foo
  print(Test.foo)
  
test()
'
```